### PR TITLE
chore: upgrade buf

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Upgrade to release buf version.

### CODE REVIEW

File: span-processing-config-service-api/src/main/proto/org/hypertrace/span/processing/config/service/v1/span_processing_config_service.proto [hunk 1]
```diff
@@ -5,7 +5,6 @@ package org.hypertrace.span.processing.config.service.v1;
 option java_multiple_files = true;
 
 import "org/hypertrace/span/processing/config/service/v1/span_filter.proto";
-import "google/protobuf/duration.proto";
```
Comment (aaron-steinfeld) [inline span-processing-config-service-api/src/main/proto/org/hypertrace/span/processing/config/service/v1/span_processing_config_service.proto:8]: only actual change, unused import caught by improved linter.

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -58,10 +58,18 @@ jobs:
...
+      - name: Setup buf
+        uses: bufbuild/buf-setup-action@v1.9.0
+        with:
+          github_token: ${{ github.token }}
+
+      - name: Lint protos
+        uses: bufbuild/buf-lint-action@v1
+
+      - name: Check for breaking changes
+        uses: bufbuild/buf-breaking-action@v1
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-build.yml:70]: These pre-packaged actions will provide better messaging/annotations directly on the PR.

**Comment by aaron-steinfeld** [comment]
Expected validate protos to fail here for two different reasons:
- The way CI is set up in this repo, it uses the new code with the CI config from main (for security purposes). So CI is trying to give the new buf config files to the old buf tool.
- Even if not (I have a commit where I switched this up), the old version of buf generates a different output for comparison.

So tl;dr ignore the build failure, we'll override.

File: buf.work.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version: v1
```
Comment (aaron-steinfeld) [inline buf.work.yaml:1]: These configs generated with `buf beta migrate-v1beta1` as described in https://docs.buf.build/configuration/v1beta1-migration-guide

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**